src/stream/src/executor/source/source_executor.rs

```diff
@@ -730,10 +803,13 @@ impl<S: StateStore> SourceExecutor<S> {
                         .updated_splits_in_epoch
                         .extend(latest_state);
 
-                    source_output_row_count.inc_by(chunk.cardinality() as u64);
+                    let card = chunk.cardinality();
+                    source_output_row_count.inc_by(card as u64);
                     let chunk =
                         prune_additional_cols(&chunk, split_idx, offset_idx, &source_desc.columns);
-                    yield Message::Chunk(chunk);
+                    if card > 0 {
+                        yield Message::Chunk(chunk);
```

Comment:
I remember that we also use empty chunk somewhere else in the connector as a hack?